lib/plugin.c

```diff
@@ -318,20 +320,47 @@ static int load_plugins(char const *dir, struct mptcpd_pm *pm)
         }
 
         errno = 0;
-        for (struct dirent const *d = readdir(ds);
-             d != NULL;
-             d = readdir(ds)) {
-                if ((d->d_type == DT_REG || d->d_type == DT_UNKNOWN)
-                    && l_str_has_suffix(d->d_name, ".so")) {
-                        char *const path = l_strdup_printf("%s/%s",
+        if (plugins_to_load) {
+
+                struct l_queue_entry const *entry = 
+                        l_queue_get_entries(plugins_to_load);
+
+                while (entry) {
+                        char *plugin_name = (char *) entry->data;
+                        char *const path = l_strdup_printf("%s/%s.so",
                                                            dir,
-                                                           d->d_name);
-                        load_plugin(path);
+                                                           plugin_name);
+
+                        if (access(path, F_OK) == 0)
```

Comment:
I think we want `R_OK` here.  According to the [access(2)](https://man7.org/linux/man-pages/man2/access.2.html) man page, The `F_OK` mode only checks for existence which is not the same as readability.